src/main/java/frc/team166/robot/subsystems/Drive.java

```diff
@@ -131,4 +170,69 @@ protected void end() {
             }
         };
     }
+
+    public Command drivetoProximity(double inches) {
+        return new SubsystemCommand("Drive Distance", this) {
+
+            //double realDistanceInches = frontLidar.getDistance(true);
+
+            @Override
+            protected void initialize() {
+                drivePidController.setSetpoint(tempestGyro.getAngle());
+                drivePidController.enable();
+                drivePidController.reset();
+            }
+
+            @Override
+            protected void execute() {
+                m_drive.arcadeDrive(Preferences.getInstance().getDouble(RobotMap.Preferences.K_P, 1), angle);
+            }
+
+            @Override
+            protected boolean isFinished() {
+                if (frontLidar.getDistance(true) <= inches) {
+                    return true;
+                } else {
+                    return false;
+                }
+
+            }
+
+            @Override
+            protected void end() {
+                drivePidController.disable();
+            }
+        };
+    }
+
+    public Command turnByDegrees(double degrees) {
+        return new SubsystemCommand("Turn " + degrees, this) {
+            @Override
+            protected void initialize() {
+                tempestGyro.reset();
+                drivePidController.reset();
+                drivePidController.enable();
+                drivePidController.setAbsoluteTolerance(3);
```

Comment:
What is 3? Why are we setting this here, but not every time we use the drivePidController? I would probably make 3 a preference, or at least a constant, and I would actually create a private method to do the reset, enable, etc. (since we have at least two commands that need to do this: hate to have to copy-paste every time)

Reply:
I believe this is a general configuration that should actually be in the initializer. As Bill said it should probably be a preference.